src/mapblock_mesh.h

```diff
@@ -161,17 +164,69 @@ class MapBlockMesh
 /*
 	This is used because CMeshBuffer::append() is very slow
 */
-struct PreMeshBuffer
+class PreMeshBuffer
 {
+private:
+	bool m_enable_shaders;
```

Comment:
Maybe this should be renamed something else that's more accurate, such as "m_has_tangents"